### DISCUSSION TITLE
How can I trigger audio play with analogue input?

### DISCUSSION THREAD

**Discussion by revlis43** [bonsai-rx/bonsai#1824] [Q&A] [answered]
Hi, guys.

I recently started to use Bonsai for mouse behavioral set up and ran onto trouble.

This might be stupid question but there is no other place to turn to.

For now, I am trying to trigger audio play with analogue signal from NI analogue input.

What I would like to do is sending trigger signal when analogue signal is bigger than certain threshold.

When tried to use "Greater Than", but error occurred with the message "Unsupported number of arguments. This node requires at least 2 input connection(s)".

I tried several nodes like "Int" or "Double" with threshold values but they were not compatible with "Greater Than".

How can I solve this problem, or is there any other options I can try?

Thanks in advance,
PG

![bonsai work flow](https://github.com/bonsai-rx/bonsai/assets/48897207/6c8c425b-3d32-4eb8-876d-1f007964a1e5)
[Sound play.zip](https://github.com/bonsai-rx/bonsai/files/15464358/Sound.play.zip)

**Comment by bruno-f-cruz** [comment]
Heya. 
Depends on exactly what behavior you are looking for, but generally you can:

![image](https://github.com/bonsai-rx/bonsai/assets/7049351/d2f3d3a9-a92d-4411-a653-53551900cab7)

**Reply by revlis43** [reply]
Hello.

Thank you for the answer!

Now I can send 'True' signal through 'Greater Than' when analogue signal is higher than threshold.

But I have ran into another trouble.

My behavior set up is about mouse pushing lever and getting reward.

Analogue input is voltage signal which linearly increases when mouse pushes lever further enough.

What I want to do is to play sound when lever is pushed further than certain distance (Threshold).

The problem is, 'False' signal occurs constantly and also triggers Audio Playback.

Can I drop the 'False' signal and only play the sound when 'True' signal comes in? (When lever is pushed further enough that analogue signal is higher than threshold)

Thank you,
PG

**Reply by bruno-f-cruz** [reply, marked as answer]
`Condition` is probably what you want. I would encourage you to go through the basics of bonsai by following the tutorials. These basic patterns (or something very close) are shown there. 
https://bonsai-rx.org/docs/tutorials/synching.html#exercise-6-recording-response-triggered-videos

**Reply by revlis43** [reply]
Thank you so much!
